BUILDING

```diff
@@ -1 +1,13 @@
-To build Ion, use the build.sh or build.py scripts in the ion/ directory.
+1. Install dependencies
+  - gyp
+  - clang
+  - libc++-dev
+  - libc++abi-dev
+
```

Comment:
Clang 3.3 definitely doesn't work.

I managed to build it on Ubuntu 14.04 (x86_64). I used Clang 3.7.1 from http://llvm.org/releases/download.html#3.7.1 . I started from a very minimal Ubuntu install (actually I ran an Ubuntu Core 14.04.3 image http://cdimage.ubuntu.com/ubuntu-core/releases/14.04/release/ under PRoot). I needed the following dependencies to build:
```
apt-get install firefox build-essential python2.7 python libgl1-mesa-dev
apt-get build-dep firefox
```
The above dependencies are probably overkill, but they did the trick.

I had to edit dev/linux.gypi to set the compiler_dir to the root of my clang directory, and I had to add my clang library directory to the library_dirs array.

This was enough to build ion. In order to actually run any of the binaries I had to add my Clang library path to LD_LIBRARY_PATH (so ld picks up libc++.so and libc++abi.so).

In theory the above should allow you to build working Ubuntu 14.04 binaries. I actually also had to build Mesa's llvmpipe software rasterizer (since I was running the binaries inside the PRoot, and my host system is an Ubuntu 12.04 box, so I don't want binaries inside the PRoot to be poking at my graphics card).

Not all of the demos worked (gearsdemo and skindemo both gave blank screens). This may have been to do with llvmpipe:
```
root@foo:~/ion/gyp-out/linux/dbg/demos# ./gearsdemo  
INFO [../../../ion/demos/utils.cc:52] Loading gear.obj
ERROR [../../../ion/gfx/renderer.cc:266] ***ION: Unable to compile vertex shader for 'Instanced gear shader vertex shader': 0:18(12): warning: extension `GL_EXT_draw_instanced' unsupported in vertex shader
0:26(14): error: syntax error, unexpected NEW_IDENTIFIER, expecting '{'

WARNING [../../../ion/gfx/renderer.cc:4437] ***ION: Attribute array contains buffer attribute 'aVertex' but the current shader program 'Instanced gear shader' does not declare or use it
WARNING [../../../ion/gfx/renderer.cc:5777] ***ION: Instanced drawing is not available. Shape: External geometry will be drawn only once.

root@foo:~/ion/gyp-out/linux/dbg/demos# ./skindemo 
INFO [../../../ion/demos/utils.cc:52] Loading head.obj
WARNING [../../../ion/gfx/renderer.cc:4437] ***ION: Attribute array contains buffer attribute 'aTexCoords' but the current shader program 'Depth shader' does not declare or use it

```

Incidentally the dbg and opt builds work, but the "prod" builds crashed Clang.